core/src/main/java/com/amplifyframework/util/UserAgent.java

```diff
@@ -235,7 +235,22 @@ public String toString() {
 
         @NonNull
         private static String sanitize(@Nullable String string) {
-            return string != null ? string : "UNKNOWN";
+            return string != null ? stripHeader(string) : "UNKNOWN";
+        }
+
+        /**
+         * Strips non-permitted characters in the header by okhttp.
+         */
+        private static String stripHeader(String value) {
+            StringBuilder builder = new StringBuilder(value.length());
```

Comment:
nit: Does it have to be a fixed length builder?

Reply:
It avoids resizing the array, since in this case we know that string size won't exceed the original length.